rasa_sdk/knowledge_base/actions.py

```diff
@@ -205,21 +224,47 @@ def _query_attribute(
             dispatcher.utter_message(template="utter_ask_rephrase")
             return [SlotSet(SLOT_MENTION, None)]
 
-        object_of_interest = self.knowledge_base.get_object(object_type, object_name)
+        if utils.is_coroutine_action(self.knowledge_base.get_object):
+            object_of_interest = await self.knowledge_base.get_object(
+                object_type, object_name  # type: ignore
+            )
+        else:
+            object_of_interest = self.knowledge_base.get_object(
+                object_type, object_name
+            )
 
         if not object_of_interest or attribute not in object_of_interest:
             dispatcher.utter_message(template="utter_ask_rephrase")
             return [SlotSet(SLOT_MENTION, None)]
 
         value = object_of_interest[attribute]
-        repr_function = self.knowledge_base.get_representation_function_of_object(
-            object_type
-        )
+        if utils.is_coroutine_action(
+            self.knowledge_base.get_representation_function_of_object
```

Comment:
should be `async` in the interface

Reply:
please also update `InMemoryKnowledgeBase`